### PR TITLE
Allow event for right header

### LINKED ISSUE #307: Event To Alter Top Right Data
It would be great to have an observer which allows to edit the data in the top right corner (document number, payment method, etc). It should be possible to delete and edit existing entries as well as adding new ones.

### PR DESCRIPTION
**Pull request by Schrank**
Refactoring, collecting written lines, add event

Fixes #307

### CODE REVIEW

**Comment by Schrank** [comment]
@sprankhub Please review _insertHeader in Abstract.php

File: src/app/code/community/FireGento/Pdf/Model/Engine/Abstract.php [hunk 1]
```diff
@@ -420,37 +407,32 @@ protected function insertAddressesAndHeader(
         $this->_insertSenderAddessBar($page);
 
         // Add head
-        // make sure that header does not overlap address bar
```
Comment (sprankhub) [inline src/app/code/community/FireGento/Pdf/Model/Engine/Abstract.php]: This was a fix we merged yesterday. Please re-add it.

File: src/app/code/community/FireGento/Pdf/Model/Engine/Abstract.php [hunk 2]
```diff
@@ -226,21 +236,20 @@ protected function Ln($height = 15)
     /**
      * Insert sender address bar
      *
-     * @param  Zend_Pdf_Page $page Current page object of Zend_Pdf
+     * @param  Zend_Pdf_Page &$page Current page object of Zend_Pdf
      *
      * @return void
      */
-    protected function _insertSenderAddessBar($page)
+    protected function _insertSenderAddessBar(&$page)
```
Comment (sprankhub) [inline src/app/code/community/FireGento/Pdf/Model/Engine/Abstract.php]: Didn't you remove all the `&` references yesterday?

File: src/app/code/community/FireGento/Pdf/Model/Engine/Abstract.php [hunk 3]
```diff
@@ -463,120 +445,85 @@ protected function _insertCustomerAddress($page, $order)
...
 
         $this->_setFontBold($page, 15);
 
-        if ($mode == 'invoice') {
-            $title = 'Invoice';
-        } elseif ($mode == 'shipment') {
-            $title = 'Packingslip';
-        } else {
+        $title = 'Invoice';
+        if ($this->getMode() === 'creditmemo') {
```
Comment (sprankhub) [inline src/app/code/community/FireGento/Pdf/Model/Engine/Abstract.php]: Why don't you use `$mode` here?

File: src/app/code/community/FireGento/Pdf/Model/Engine/Abstract.php (hunk 2 above)
Comment (sprankhub) [reply src/app/code/community/FireGento/Pdf/Model/Engine/Abstract.php]: Although you should not remove all:

    2017-10-14T16:28:01+00:00 ERR (3): Warning: Declaration of FireGento_Pdf_Model_Engine_Abstract::insertLogo($page, $store = NULL) should be compatible with Mage_Sales_Model_Order_Pdf_Abstract::insertLogo(&$page, $store = NULL)  in /var/www/magento193/.modman/firegento-pdf/src/app/code/community/FireGento/Pdf/Model/Engine/Abstract.php on line 0

**Pull request merged**